### ISSUE TITLE
how to reproduce results on the paper?

### ISSUE THREAD

**Issue by tackgeun** [jwyang/lr-gan.pytorch#4]
Hi, I'm very impressed by your work LR-GAN. So, I'm trying to reproduce results (CUB200, CIFAR10) on the paper, but it totally doesn't learn at all.

- When I train the algorithm with given parameters in the README (CUB200, CIFAR10)
    - discriminator loss goes to zero, while generator can't generate any meaningful pattern with fixed seed.
    - I've tried several difference learning rates for discriminator and generation, it didn't work also.
- If I don't use foreground & mask generator and STN, time step = 1, it produces good results like DCGAN after training.

Could you give some information for reproducing the result? I'm using pyTorch with version '0.2.0_2' on python 3.6 and nvidia 367.57 with cuda 8.0.

Thank you.

**Comment by jwyang** [comment]
Hi, I updated this code after checking on all datasets. I will try it out on my side to see what it will happen.

**Comment by tackgeun** [comment]
Thank you in advance. After posting, I've found that the problem is related with STNM. I guess that the compile of STNM hasn't been done properly. I'll also report when I find the solution.

**Comment by jwyang** [comment]
great, let me know what you get when you update stnm

**Comment by tackgeun** [comment]
I got realistic CUB-200 images by changing STNM into two STNs(https://github.com/fxia22/stn.pytorch) for foreground image and mask. But I not checked other stuffs in the paper.
Because I can't code CUDA, I can't debug deeper at now.

**Comment by jwyang** [comment]
how does the result look like?

**Comment by tackgeun** [comment]
http://cvlab.postech.ac.kr/~tackgeun/LRGAN.dSTN/img.html
Generated samples are relatively darker than the paper.
The out-of-center region of generated foreground images is darker than the paper.

**Comment by jwyang** [comment]
I think it is a problem of displaying. You need to convert the range of images to [0, 255] before displaying. As you can see, the real image are also dark. That explains.

**Comment by tackgeun** [comment]
Thank you for advice. I take a reversed pre-processing step in saving a resulting image.
